uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/AbsenceTypeTest.php

```diff
@@ -22,6 +22,13 @@ public function setUpHeadless() {
     return \Civi\Test::headless()->installMe(__DIR__)->apply();
   }
 
+  public function setUp() {
+    // We delete everything two avoid problems with the default absence types
+    // created during the extension installation
+    $tableName = CRM_HRLeaveAndAbsences_BAO_AbsenceType::getTableName();
```

Comment:
I see there is no need to do this ,so in this test method :

testUpdatingATypeWithAddPublicHolidayToEntitlementShouldNotTriggerErrorAboutHavingAnotherTypeWithItSelected

You can retrieve the default absence type holiday/vacation and get its ID and then perform
  $this->updateBasicType($entity1->id, ['add_public_holiday_to_entitlement' => true]); on it.